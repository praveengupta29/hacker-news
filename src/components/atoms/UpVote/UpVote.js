import React from 'react';
import PropTypes from 'prop-types';

import Anchor from '../Anchor';
import './UpVote.scss';

const UpVote = ({ voteFeed, objectId }) => {
  return (
    <Anchor
      handleLinkClick={e => {
        e.preventDefault();
        voteFeed({
          objectId,
          voted: true,
        });
      }}
      className="up-vote-link"
    >
      <div className="up-vote-arrow" />
    </Anchor>
  );
};

UpVote.propTypes = {
  objectId: PropTypes.string.isRequired,
  voteFeed: PropTypes.func.isRequired,
};

export default UpVote;
