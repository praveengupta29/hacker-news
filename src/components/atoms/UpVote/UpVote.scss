@import '../../../styles/utils/colors';

.up-vote-link {
  margin-left: 6px;
}

.up-vote-arrow {
  background: url('../../../images/grayarrow2x.png') center center no-repeat
    $white;
  height: 10px;
  width: 10px;
}
