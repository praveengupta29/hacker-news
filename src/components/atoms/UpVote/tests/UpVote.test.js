import React from 'react';
import { shallow } from 'enzyme';
import UpVote from '../UpVote';

describe('UnVote component', () => {
  const props = {
    voteFeed: () => {},
    objectId: '6382672',
  };

  it('should render correctly and match snapshot', () => {
    const component = shallow(<UpVote {...props} />);
    expect(component).toMatchSnapshot();
  });
});
