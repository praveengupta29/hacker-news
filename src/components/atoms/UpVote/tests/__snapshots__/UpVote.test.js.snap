// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnVote component should render correctly and match snapshot 1`] = `
<Anchor
  className="up-vote-link"
  handleLinkClick={[Function]}
  target="_self"
  to=""
>
  <div
    className="up-vote-arrow"
  />
</Anchor>
`;
